libs/navigation/src/drivers/magento/queries/fragments/category-node.spec.ts

```diff
@@ -195,7 +195,23 @@ describe('Navigation | Driver | Magento | getCategoryNodeFragment', () => {
         data: depth3NavigationTree
       })
     });
-  });
+	});
+	
+	it('should return the expected fragment', () => {
+		const expectedFields = [
+			'id',
+			'level',
+			'name',
+			'include_in_menu',
+			'products',
+			'children_count',
+			'children'
+		];
+
+		(<any>getCategoryNodeFragment(3).definitions[0]).selectionSet.selections.forEach((selection, index) => {
+			expect(selection.name.value).toEqual(expectedFields[index]);
+		});
```

Comment:
This spec doesn't test the behavior of the function, only its particular implementation. Even if the function was working as it should it wouldn't pass this test if it used fragments.

Reply:
Do you mean "Even if the function had been working as it should..."? Because it does work now. While I agree that this test wouldn't have worked if the fragment spreading had been working as expected, it does work as the function stands now. It seems to me that this test does test the behavior of the function, because the way the function was implemented changed. Just because it wouldn't have passed before doesn't mean it doesn't accomplish its intended purpose. That being said, I also would agree that ideally this test would pass/fail for both implementations, and I'm really not sure how to write that test.

Reply:
Sorry I meant my previous implementation. Even if my implementation wasn't causing Magento to throw errors it wouldn't pass this test. So this test doesn't correctly describe the behavior of the function, only this particular implementation.

Hypothetically, let's say that Magento fixes that bug (if indeed it is a bug) and we can move back to my fragment implementation. We would have to remove your test case. Ideally, tests should only change when the feature requirements change. Since the "black box" behavior is not changing, swapping equivalent implementations should not fail tests. This indicates to me that this test case does not belong.

>I'm really not sure how to write that test

I don't think its possible. I think this is Magento-specific behavior and is a bug.